Source/WebCore/contentextensions/DFABytecodeCompiler.cpp

```diff
@@ -510,7 +510,6 @@ void DFABytecodeCompiler::compile()
     // Populate m_maxNodeStartOffsets with a worst-case index of where the node would be with no branch compaction.
     // Compacting the branches using 1-4 byte signed jump distances should only make nodes closer together than this.
     ASSERT(m_maxNodeStartOffsets.isEmpty());
-    m_maxNodeStartOffsets.clear();
```

Comment:
This makes us reuse a buffer rather than deallocating and reallocating each time. Is there any effect of this other than improving performance?

Reply:
I assume your concern is that the allocated buffer may now be larger than we need, thus wasting memory? I don't see other drawbacks here.

I looked into this concern and `m_maxNodeStartOffsets` is owned by the `DFABytecodeCompiler`. `DFABytecodeCompiler` objects are always held on the stack so I don't think the buffer being potentially larger than the actual data is an issue here. 

I don't see `m_maxNodeStartOffsets` being passed / transferred to any other object either.

Based on this, I *think* my change is strictly a perf win? Please let me know if I missed something.

Reply:
It seemed possible, albeit unlikely, that there was some sort of security implication or something else subtle. I’m glad that you don’t think so.